modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -1782,11 +1821,70 @@ private RuntimeException convertThrowable(Throwable th) {
         return new IgniteException(th);
     }
 
-    /**
-     * Register the new meta storage listener for changes in the rebalance-specific keys.
-     */
-    private void registerRebalanceListeners() {
-        metaStorageMgr.registerPrefixWatch(ByteArray.fromString(PENDING_ASSIGNMENTS_PREFIX), new WatchListener() {
+    private WatchListener createDistributionZonesDataNodesListener() {
```

Comment:
Javadocs, here and below for new methods that you've introduced

Reply:
Added.